Added sample of NodeJS application that requires DB

sleshchenko opened the pull request:
This PR adds a sample of NodeJS application that requires DB

sleshchenko committed:
Add sample of NodeJS application that requires DB

mshaposhnik commented on samples/web-nodejs-with-db-sample/devfile.yaml:
```diff
@@ -0,0 +1,26 @@
+specVersion: 0.0.1
+name: multiple-k8s-tools
```
Maybe name it smthng like `nodejs-with-db` ?

sleshchenko replied:
Good catch.

mshaposhnik commented:
It seems we should use `devfile.yml` not `.yaml` (https://github.com/eclipse/che/issues/12837)

mshaposhnik reviewed (approved):
other is LGTM

sleshchenko committed:
fixup! Add sample of NodeJS application that requires DB

sleshchenko committed:
Fix YAML file extension in theia-hello-world-frontend-plugin

sleshchenko commented:
> It seems we should use devfile.yml not .yaml (eclipse/che#12837)

You're right. Fixed.
YAML file extension is also fixed in `theia-hello-world-frontend-plugin` sample

l0rd commented:
Nice work @sleshchenko 

Are the services and routes of the 2 YAMLs used in the Che workspace? I mean are we creating a workspace server corresponding to the nodejs route?

I am asking because if we do not use these kubernetes resources in a Che workspace we may want to filter them out (and make both tools of type kubernetes). And anyhow routes should be automatically setup once https://github.com/eclipse/che/issues/12061 is completed.

sleshchenko commented:
@l0rd 

> Are the services and routes of the 2 YAMLs used in the Che workspace? I mean are we creating a workspace server corresponding to the nodejs route?

All objects are created. Mongo DB service is used, as you can see here, application access to database via `mongo` host 
https://github.com/sleshchenko/devfile/blob/nodejs-app-with-db/samples/web-nodejs-with-db-sample/devfile.yml#L26

As I showed in my Demo https://youtu.be/e4UrtbTmV7M, route is not accessible for a user from Theia, but he is able to get it from OpenShift Console.

I like the idea to make sample usable for Kubernetes infra too, but until https://github.com/eclipse/che/issues/12061 is not complited, removing route is not a good thing to do because user won't be able to access NodeJS Application.
Also, using OpenShift tools shows an ability to use OpenShift template with default parameters, maybe it should be done as a dedicated sample.

WDYT? Do you think it should be removed now or after resolving of https://github.com/eclipse/che/issues/12061?

l0rd commented:
Thank you @sleshchenko for the explanation I think we can merge it like that
